### ISSUE TITLE
Add type definitions for main client

### ISSUE THREAD

**Issue by ryanblock** [architect/aws-lite#2]
Right now it's looking like this:
```js
/**
 * @param {string} [options.host] Manually specify a hostname
 * @param {string} [options.endpoint=/] API endpoint path; should not include domain
 * @param {string} [options.region] Change the AWS service region for a single request
 * @param {string|object|array} [options.payload] JSON-serialized data, or object or array to be JSON-serialized
 * @param {string|object|array} [options.json] Alias of options.payload
 * @param {string|object|array} [options.body] Alias of options.payload
 * @param {string|object|array} [options.data] Alias of options.payload
 * @param {object} [options.headers] Request headers to overlay
 * 
 * @returns {Promise<any>} Result or error
 */
```

**Comment by tbeseda** [comment]
I've got a local prototype working of how `aws-lite/client` can declare its own `module` and plugins can extend the global returned by that module.
![image](https://github.com/architect/aws-lite/assets/15697/7b7ed19e-49a5-462d-a817-1eb3f692545f)
I know screenshots aren't the best way to show code, something more concrete soon

**Comment by ryanblock** [comment]
Shipped in 0.8.0!
